### PR TITLE
[No QA] Make schema validations async

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This is just a minor way to speed up PR checks.

### Fixed Issues
$ n/a

### Tests
1. Run `npm i`
1. Run `npm run gh-actions-validate`. Verify that the script runs successfully.
1. Remove a required key from an action metadata file.
1. Run `npm run gh-actions-validate`. Verify that the script catches the error, reports it successfully, and exits with a non-zero status code.
1. Restore the action metadata file.
1. Remove a required key from a workflow file.
1. Run `npm tun gh-actions-validate`. Verify that the script catches the error, reports it successfully, and exists with a non-zero status code.

### CODE REVIEW

**Review by aldo-expensify — CHANGES_REQUESTED** [review]
Cool update, it runs much faster, but I think there is something to correct.

The exit code is wrong sometimes... if you only remove the "on" and "name" of cla.yml, you can see it detects the invalid file, but the exit code is 0. If you do this same change in `main`, the exit code is 1.

image

If I mess up even more the file, like removing the `jobs` key without removing the descendants, then I get an exit code 1

image

Screen Shot 2022-07-11 at 10 37 56 AM

**Comment by roryabraham** [comment]
This is updated now, seems to work as expected now

File: .github/scripts/validateActionsAndWorkflows.sh [hunk 1]
```diff
@@ -32,20 +47,17 @@ done
...
 # Wait for the background builds to finish
 for PID in ${ASYNC_PROCESSES[*]}; do
-  if ! wait $PID; then
-    EXIT_CODE=$?
+  wait $PID
+  RESULT=$?
+  if [[ $RESULT != 0 ]]; then
+    EXIT_CODE=$RESULT
   fi
 done
```
Comment (aldo-expensify) [inline .github/scripts/validateActionsAndWorkflows.sh:55]: I think this is what fixed the bug, right?

Comment (roryabraham) [reply .github/scripts/validateActionsAndWorkflows.sh:55]: Yeah. When you have an `if` statement that in-and-of-itself is a command and so it changes the value of `$?`.

**Review by aldo-expensify — APPROVED** [review]

**Comment by roryabraham** [comment]
This is working well

**Pull request merged**